src/test/java/com/iota/iri/service/tipselection/impl/WalkValidatorImplTest.java

```diff
@@ -251,9 +241,102 @@ public void failOnInconsistency() throws Exception {
         milestoneTracker.latestSolidSubtangleMilestoneIndex = Integer.MAX_VALUE;
 
         WalkValidatorImpl walkValidator = new WalkValidatorImpl(tangle, ledgerValidator, transactionValidator,
-                milestoneTracker, 15, Integer.parseInt(Configuration.BELOW_MAX_DEPTH_LIMIT),
-                Integer.parseInt(Configuration.WALK_VALIDATOR_CACHE));
+                milestoneTracker, 15, Integer.parseInt(Configuration.BELOW_MAX_DEPTH_LIMIT));
         Assert.assertFalse("Validation succeded but should have failed due to inconsistent ledger state",
                 walkValidator.isValid(hash));
     }
+
+    @Test
+    public void dontMarkWrongTxsAsBelowMaxDepth() throws Exception {
+        final int maxAnalyzedTxs = Integer.parseInt(Configuration.BELOW_MAX_DEPTH_LIMIT);
+        TransactionViewModel tx1 = TransactionTestUtils.createBundleHead(0);
+        tx1.store(tangle);
+        tx1.setSnapshot(tangle, 92);
+
+        TransactionViewModel txBad = TransactionTestUtils.createBundleHead(0);
+        txBad.store(tangle);
+        txBad.setSnapshot(tangle, 10);
+
+        TransactionViewModel tx2 = new TransactionViewModel(TransactionViewModelTest.getRandomTransactionWithTrunkAndBranch(tx1.getHash(), tx1.getHash()),
```

Comment:
Minor thing: can we give more descriptive names to tx2 / tx3... ? Otherwise test is not that readable.

Reply:
tx1 => txGood
tx2 => txPointsToGood
tx4 => txPointsToBad